File: leetcode/merge-k-sorted-lists/merge-k-sorted-lists.go
package mergeksortedlists

import "math"

type ListNode struct {
	Val  int
	Next *ListNode
}

func mergeKLists(lists []*ListNode) *ListNode {
	var index, min int
	index, min = -1, math.MaxInt32
	for i, node := range lists {
		if node != nil && node.Val < min {
			index = i
			min = node.Val
		}
	}
	if index == -1 {
		return nil
	}
	head := lists[index]
	lists[index] = lists[index].Next
	head.Next = mergeKLists(lists)
	return head
}

File: leetcode/merge-k-sorted-lists/merge-k-sorted-lists.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def mergeKLists(self, lists: List[ListNode]) -> ListNode:
        head = node = ListNode()
        while True:
            lists = [val for val in lists if val is not None]
            obj = min(lists, key=lambda x: x.val, default=None)
            if obj is None:
                break
            index = lists.index(obj)
            node.next = ListNode(obj.val)
            node = node.next
            lists[index] = lists[index].next
        return head.next
